homeassistant/components/satel_integra/__init__.py

```diff
@@ -92,63 +101,118 @@ def is_alarm_code_necessary(value):
+            "domain": DOMAIN,
+            "integration_title": "Satel Integra",
+        },
     )
 
-    controller = AsyncSatel(host, port, hass.loop, zones, monitored_outputs, partitions)
 
-    hass.data[DATA_SATEL] = controller
+async def async_setup_entry(hass: HomeAssistant, entry: SatelConfigEntry) -> bool:
+    """Set up  Satel Integra from a config entry."""
+
+    host = entry.data[CONF_HOST]
+    port = entry.data[CONF_PORT]
+
+    # Make sure we initialize the Satel controller with the configured entries to monitor
+    partitions = [
+        subentry.data[CONF_PARTITION_NUMBER]
+        for subentry in filter(
+            lambda subentry: subentry.subentry_type == SUBENTRY_TYPE_PARTITION,
+            entry.subentries.values(),
+        )
+    ]
+
+    zones = [
+        subentry.data[CONF_ZONE_NUMBER]
+        for subentry in filter(
+            lambda subentry: subentry.subentry_type == SUBENTRY_TYPE_ZONE,
+            entry.subentries.values(),
+        )
+    ]
+
+    outputs = [
+        subentry.data[CONF_OUTPUT_NUMBER]
+        for subentry in filter(
+            lambda subentry: subentry.subentry_type == SUBENTRY_TYPE_OUTPUT,
+            entry.subentries.values(),
+        )
+    ]
+
+    switchable_outputs = [
+        subentry.data[CONF_SWITCHABLE_OUTPUT_NUMBER]
+        for subentry in filter(
+            lambda subentry: subentry.subentry_type == SUBENTRY_TYPE_SWITCHABLE_OUTPUT,
+            entry.subentries.values(),
+        )
+    ]
+
+    monitored_outputs = outputs + switchable_outputs
+
+    controller = AsyncSatel(host, port, hass.loop, zones, monitored_outputs, partitions)
 
     result = await controller.connect()
 
     if not result:
-        return False
+        raise ConfigEntryNotReady("Controller failed to connect")
+
+    entry.runtime_data = SatelData(controller)
 
     @callback
     def _close(*_):
         controller.close()
 
-    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, _close)
+    entry.async_on_unload(entry.add_update_listener(update_listener))
 
-    _LOGGER.debug("Arm home config: %s, mode: %s ", conf, conf.get(CONF_ARM_HOME_MODE))
-
-    hass.async_create_task(
-        async_load_platform(hass, Platform.ALARM_CONTROL_PANEL, DOMAIN, conf, config)
-    )
-
-    hass.async_create_task(
-        async_load_platform(
-            hass,
-            Platform.BINARY_SENSOR,
-            DOMAIN,
-            {CONF_ZONES: zones, CONF_OUTPUTS: outputs},
-            config,
-        )
-    )
+    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, _close)
 
-    hass.async_create_task(
-        async_load_platform(
-            hass,
-            Platform.SWITCH,
-            DOMAIN,
-            {
-                CONF_SWITCHABLE_OUTPUTS: switchable_outputs,
-                CONF_DEVICE_CODE: conf.get(CONF_DEVICE_CODE),
-            },
-            config,
-        )
+    await hass.config_entries.async_forward_entry_setups(
+        entry,
+        PLATFORMS,
```

Comment:
```suggestion
        PLATFORMS
```
If you remove the comma, ruff will format it nicer